src/telliot_core/contract/contract.py

```diff
@@ -107,15 +117,38 @@ async def write(
             contract_function = self.contract.get_function_by_name(func_name)
             transaction = contract_function(**kwargs)
 
-            built_tx = transaction.buildTransaction(
-                {
-                    "from": acc.address,
-                    "nonce": acc_nonce,
-                    "gas": gas_limit,
-                    "gasPrice": self.node.web3.toWei(gas_price, "gwei"),
-                    "chainId": self.node.chain_id,
-                }
-            )
+            # start tx dict with static elements
+            tx_dict = {
+                "from": acc.address,
+                "nonce": acc_nonce,
+                "gas": gas_limit,
+                "chainId": self.node.chain_id,
+            }
+
+            # if legacy gas price is not None, use it
```

Comment:
This stretch of logic (129-151) is pretty hard for me to follow what's going on - which to me, signals a need for unit tests to demonstrate the different options.

Reply:
Same, here's a possible edit:
```python
if legacy_gas_price:
    tx_dict['gasPrice'] = blah
# Transaction type 2 (EIP1559)
else:
    if max_fee_per_gas:
        tx_dict['maxFeePerGas'] = blah
    if max_priority_fee_per_gas:
        tx_dict['maxPriorityFeePerGas'] = blah
```
I think it should be able to override both max fee and priority fee. Also, we should remove this bit:
```python
# else (if none are provided) estimate max priority fee with rpc node
else:
    tx_dict["maxPriorityFeePerGas"] = self.node.web3.eth.max_priority_fee
```